applications/MultilevelMonteCarloApplication/python_scripts/cmlmc_utilities.py

```diff
@@ -0,0 +1,781 @@
+        r2   : Cost increase final iterations C-MLMC
+        '''
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "k0" : 0.1,
+            "k1" : 0.1,
+            "r1" : 1.25,
+            "r2" : 1.15,
+            "tol0" : 0.25,
+            "tolF" : 0.1,
+            "cphi" : 1.0,
+            "number_samples_screening" : 25,
+            "Lscreening" : 2,
+            "Lmax" : 4,
+            "mesh_refinement_coefficient" : 2,
+            "initial_mesh_size" : 0.5,
+            "minimum_add_level" : 6.0,
+            "splitting_parameter_max" : 0.9,
+            "splitting_parameter_min" : 0.1
+        }
+        """)
+        self.settings = custom_settings
+        '''warning if initial_mesh_size parameter not set by the user'''
+        if not (self.settings.Has("initial_mesh_size")):
+            print("\n ######## WARNING : initial_mesh_size parameter not set ---> using defalut value 0.5 ########\n")
+        '''validate and assign default parameters'''
+        self.settings.ValidateAndAssignDefaults(default_settings)        
+        '''current_number_levels : number of levels of current iteration'''
+        self.current_number_levels = self.settings["Lscreening"].GetInt()
+        '''previous_number_levels : number of levels of previous iteration'''
+        self.previous_number_levels = None
+        '''number_samples : total number of samples at current iteration'''        
+        self.number_samples = [self.settings["number_samples_screening"].GetInt() for i in range (self.settings["Lscreening"].GetInt()+1)]
+        '''difference_number_samples : difference between number of samples of current and previous iterations'''
+        self.difference_number_samples = None
+        '''previous_number_samples : total number of samples of previous iteration'''
+        self.previous_number_samples = None
+        '''rates_error : dictionary containing the values of the parameters
+        calpha : coefficient of the function maximizing bias
+        alpha  : exponent of the function maximizing bias
+        cbeta  : coefficient of the function maximizing statistical error
+        beta   : exponent of the function maximizing statistical error
+        cgamma : coefficient of the function maximizing cost
+        gamma  : exponent of the function maximizing cost
+        '''
+        self.rates_error = {"calpha":None, "alpha":None, "cbeta":None, "beta":None, "cgamma":None, "gamma":None}
+        '''mesh_parameters : reciprocal of minimal mesh size'''        
+        self.mesh_parameters = []
+        '''size_mesh : minimal mesh size'''
+        self.sizes_mesh = []
+        '''BayesianVariance : Bayesian variance'''
+        self.BayesianVariance = []
+        '''number_iterations : theoretical number of iterations the MLMC algorithm will perform'''
+        self.number_iterations_iE = None
+        '''convergence : boolean variable defining if MLMC algorithm is convergenced'''
```

Comment:
typo convergenced

Reply:
Shouldn't be better "convergence"? Or it is a Kratos standard to use"convergenced"?